stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/socket/ChatSocket.kt

```diff
@@ -150,7 +151,7 @@ internal open class ChatSocket constructor(
             state<State.Connected> {
                 onEnter {
                     if (it is Event.WebSocket.OnConnectedEventReceived) {
-                        healthMonitor.start()
+                        healthMonitor.ack()
```

Comment:
The logic below is not correct. If we receive message events (or other events) with 1 second interval the `checkCallback` will never be invoked and we will be eventually disconnected from the server.

```
  open fun onEvent(event: ChatEvent) {
      healthMonitor.ack()
      callListeners { listener -> listener.onEvent(event) }
  }
```

Reply:
Also, after disconnect. It doesn't reconnect and stays permanently disconnected.

Reply:
Good catch!!!!
It was on that way from a long time ago, and maybe some connection issues we had previously were because of that.
Fixing it